server/lib/tuist_web/user_socket.ex

```diff
@@ -0,0 +1,38 @@
+defmodule TuistWeb.UserSocket do
+  use Phoenix.Socket
+
+  alias Tuist.Authentication
+
+  channel("qa_logs:*", TuistWeb.QALogChannel)
+
+  def connect(%{"token" => token}, socket, _connect_info) do
+    case authenticate_socket(token) do
+      {:ok, subject} ->
+        socket = assign(socket, :current_subject, subject)
+        {:ok, socket}
+
+      {:error, _reason} ->
+        :error
+    end
+  end
+
+  def connect(_params, _socket, _connect_info), do: :error
+
+  def id(socket) do
+    case socket.assigns do
+      # NOTE: This is probably not unique enough? As in, one account can run multiple tests at the same time?
```

Comment:
that is correct – this is gonna be actually quite likely, so I'd suggest resolving this. We can only assume the QA run ID will be unique.

Reply:
Can the client include the QA run ID and then read it from here to include it in the id?